proxy/nasshp/nassh.go

```diff
@@ -326,18 +326,26 @@ func (np *NasshProxy) allow(r *http.Request, w http.ResponseWriter, sid, hostpor
 		logid = LogId(sid, r, hostport, creds)
 	}
 	if np.filter != nil {
-		verdict := np.filter("tcp", hostport, creds)
-		if HostIp(hostport).IsUrl() {
-			for _, u := range HostIp(hostport).Resolve() {
+		host, port, err := net.SplitHostPort(hostport)
+		if err != nil {
+			goto DENY
```

Comment:
I would avoid the goto, and differentiate the error messages? So if we ever land here, we can tell what the problem is? Also, we avoid the goto, considered harmful and evil (except in C code for cleanup).

```
if err != nil {
  np.log.Infof("%s could not split %s in filter validation", logid, hostport)
  http.Error(w, fmt.Sprintf("Something fishy in your request, cannot apply ACLs, rejected)
}